tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -125,12 +317,13 @@ async Task<string[]> InternalSearchCommitAsync(byte[] state, bool finalTry)
         {
             var request = _apiRequestFactory.Create(_searchCommitsUrl);
             request.AddHeader(ApiKeyHeader, _settings.ApiKey);
+            request.AddHeader(HttpHeaderNames.TraceId, _id);
```

Comment:
Out of curiosity, in which case you need an appkey in addition to the apikey.

Reply:
we need it for two transactions, gets the configuration for ITR and to query the tests to be skipped.